<!DOCTYPE html>
<html lang="en-us"><head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=57852&amp;path=livereload" data-no-instant defer></script>
<title>Introduction to protein design: Overview and history - Stephen Malina</title>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, viewport-fit=cover">
<meta name="description"
    content="A comprehensive introduction to the history and methods of protein design, from early synthesis techniques to directed evolution and modern approaches. ">
<link rel="canonical" href="http://localhost:57852/post/2022-12-18-annotated-ml-protein-design-bibliography/" />



<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/modern-normalize/1.1.0/modern-normalize.min.css" crossorigin="anonymous" referrerpolicy="no-referrer" />



<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
<link rel="preload" as="style"
      href="https://fonts.googleapis.com/css?family=Noto+Serif+SC|Noto+Emoji&display=swap" />
<link rel="stylesheet"
      href="https://fonts.googleapis.com/css?family=Noto+Serif+SC|Noto+Emoji&display=swap"
      media="print" onload="this.media='all'" />
<noscript>
<link rel="stylesheet"
      href="https://fonts.googleapis.com/css?family=Noto+Serif+SC&display=swap" />
</noscript>


  
  
    
      <link rel="stylesheet" href="http://localhost:57852/css/hugo-tufte.min.css">
    
  



  
  
    
      <link rel="stylesheet" href="http://localhost:57852/css/hugo-tufte-options.min.css">
    
  


<link rel="stylesheet" href="http://localhost:57852/css/hugo-tufte-override.css">

</head>
<body>


<article id="main">
  <section>
<h1 class="content-title">Introduction to protein design: Overview and history&nbsp;:: Draft</h1></section>

  

  <section><p><strong>Status</strong>: Work in progress.</p>
<h2 id="introduction">
Introduction
<a href="#introduction" class="heading-anchor">#</a>
</h2>
<p>Protein design is the art of creating new protein sequences and structures that have specific properties or functions. Proteins are made up of strings of amino acids and are essential to many biological processes. They serve a wide range of roles in the body, including catalyzing chemical reactions, transporting molecules, and providing structural support. Proteins have numerous practical therapeutic applications, such as viral vectors for gene therapy, mini proteins for targeting specific molecules, and improved antibodies for fighting diseases. Proteins also have applications outside of therapeutics. Enzymes are used in a huge and growing fraction of industrial chemical processes and also play a key role in sustainable potential replacements for existing processes.</p>
<p>The goal of protein design is to create new proteins that optimize for certain properties or functions. This is hard because proteins are complex molecules with intricate structures and functions that are not fully understood. To design new proteins, we must first understand the relationship between a protein&rsquo;s sequence and its structure and function.</p>
<p>Machine-guided protein design involves using machine learning algorithms to help design new proteins based on existing protein sequences and their associated properties. This can be framed as an ML problem, where we have a dataset of protein sequences and their properties, and the goal is to design a new set of protein sequences that improve upon one or more properties. By leveraging the power of machine learning, we can more efficiently and effectively design new proteins that have the desired properties or functions.</p>
<p>Especially with recent success in structure prediction, progress and interest in ML for protein design has accelerated over the past few years. This post series is planned to have three parts. Part 1 (this post) covers the history of protein design as a field prior to the introduction of ML and statistical models. Part 2 will cover the introduction of machine learning into the protein world more broadly, and then part 3 will cover how ML is being used for protein design specificaly. The target audience is people who have an ML background, basic familiarity with biology, but limited exposure to protein design or function prediction.</p>
<h2 id="short-history-of-protein-synthesis-and-design">
Short history of protein synthesis and design
<a href="#short-history-of-protein-synthesis-and-design" class="heading-anchor">#</a>
</h2>
<h3 id="history-of-protein-synthesis">
History of protein synthesis
<a href="#history-of-protein-synthesis" class="heading-anchor">#</a>
</h3>
<p>The field of protein design has a long history dating back to the beginning of the 20th century, when the first proteins were artificially synthesized. Early efforts in protein design focused on synthesizing small proteins and peptides using chemical methods. The first known case of peptide synthesis happened alongside the creation of the word peptide in 1901 when Emil Fischer and E Fourneau synthesized glycylglycine from glycine (via hydrolysis) (<a href="https://www.ias.ac.in/article/fulltext/reso/016/07/0640-0647">One Hundred Years of Peptide Chemistry</a>). The next landmark in peptide synthesis didn&rsquo;t happen until the 1950s when Vigneaud isolated and chemically synthesized oxytcoin.</p>
<p>Peptide synthesis was a significant advance in the field of biochemistry when it was developed in the 1950s but synthesizing proteins, which are much larger and more complex molecules, required a different approach. In the 1950s, techniques for synthesizing proteins on a large scale did not exist, and the field of molecular biology was still in its early stages. Instead, the discovery and refinement of molecular cloning techniques in the 1970s and 1980s played a crucial role in the development of protein synthesis methods. Molecular cloning allows for the production of large amounts of a specific gene or DNA sequence, which can then be used to synthesize the corresponding protein. <label for="sidenote-1" class="margin-toggle sidenote-number">(1)</label>
<input type="checkbox" id="sidenote-1" class="margin-toggle"/>
<span class="sidenote">
<span class="sidenote-number">(1)</span>Genentech was founded to commercialize this technology and famously used it to synthesize insulin in bacteria. <a href="https://www.amazon.com/Genentech-Beginnings-Sally-Smith-Hughes/dp/022604551X/ref=asc_df_022604551X/?tag=hyprod-20&amp;linkCode=df0&amp;hvadid=312165853622&amp;hvpos=&amp;hvnetw=g&amp;hvrand=14551752941317260001&amp;hvpone=&amp;hvptwo=&amp;hvqmt=&amp;hvdev=c&amp;hvdvcmdl=&amp;hvlocint=&amp;hvlocphy=9067609&amp;hvtargid=pla-453362017359&amp;psc=1&amp;tag=&amp;ref=&amp;adgrpid=60258872297&amp;hvpone=&amp;hvptwo=&amp;hvadid=312165853622&amp;hvpos=&amp;hvnetw=g&amp;hvrand=14551752941317260001&amp;hvqmt=&amp;hvdev=c&amp;hvdvcmdl=&amp;hvlocint=&amp;hvlocphy=9067609&amp;hvtargid=pla-453362017359">Genentech: The Beginnings of Biotech</a> covers the history of Genentech, in the process describing the early history of molecular cloning.
</span>
These advances paved the way for the development of methods for producing proteins on a large scale, including recombinant DNA technology and expression systems.</p>
<h3 id="methods-for-protein-design-and-characterization">
Methods for protein design and characterization
<a href="#methods-for-protein-design-and-characterization" class="heading-anchor">#</a>
</h3>
<h4 id="directed-evolution">
Directed evolution
<a href="#directed-evolution" class="heading-anchor">#</a>
</h4>
<p>Directed evolution was the first method for designing novel proteins. As the name suggests, it harnesses selection pressure to evolve proteins with desired properties. In directed evolution, a library of protein variants is generated and exposed to a selection or screening process that selects for the variants with the desired properties. The selected variants are then used to generate a new library, and the process is repeated until the desired property is optimized. While its experimental history dates back to the 1960s with the <a href="https://en.wikipedia.org/wiki/Spiegelman%27s_Monster">Spiegelman&rsquo;s Monster experiment</a>, it wasn&rsquo;t until the 1990s that Francis Arnold and her team at Caltech showed that the process could be done inside bacteria and <a href="(https://web.archive.org/web/20170430132138/https://www.che.caltech.edu/groups/fha/publications/Arnold_ACR_1998.pdf)">employed it for designing improved enzymes</a>. This approach was initially low-throughput, as it relied on manual screening or selection of the protein variants.</p>
<p>Directed evolution has since been used to design proteins for a wide range of applications, <a href="https://en.wikipedia.org/wiki/Directed_evolution#Applications">including</a> including antibody design and industrial processes. It has also been used to design <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3147008/">proteins for bioremediation and environmental applications</a>, such as the cleanup of oil spills and the removal of pollutants from water.</p>
<h4 id="deep-mutational-scanning">
Deep mutational scanning
<a href="#deep-mutational-scanning" class="heading-anchor">#</a>
</h4>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4410700/">Deep mutational scanning</a> is a method of systematically varying the amino acids in a protein sequence and measuring the effect on the protein&rsquo;s function. It involves generating a large library of protein variants with single amino acid substitutions at every position in the protein and measuring the effect of each variant on the protein&rsquo;s function.</p>
<p>Deep mutational scanning was enabled by the advent of cheap high-throughput screening that could be used to design variant-level assays of protein properties. The basic principle behind these assays is to find some way of using sequencing as a proxy for the property of interest by coupling the characteristic to abundance by way of selection. Many deep mutational scans (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3991786/">example</a>) accomplish this by coupling the property of interest to flourescence, which then gets used to sort cells or molecules into buckets. Then sequencing is used to identify which mutants are in each bucket (potentially at different abundance levels).</p>
<p>Deep mutational scanning was first introduced in the early 2010s and has become a widely used method for studying protein function and for improving protein design. It has been used to identify the key residues in proteins that are responsible for their function and to design proteins with improved or novel functions. While deep mutational scanning itself only characterizes the fitness of mutatns vs. designing them, the data it generates can immediately be used for design. For example, the simplest deep mutational scan-based design strategy is to take the top N variants from a scan and treat those as newly designed proteins. Deep mutational scanning also allows for enhanced directed evolution by expanding the size of the population we can</p>
<h3 id="structure-based-design">
Structure-based design
<a href="#structure-based-design" class="heading-anchor">#</a>
</h3>
<h3 id="ml-guided-protein-design">
ML-guided protein design
<a href="#ml-guided-protein-design" class="heading-anchor">#</a>
</h3>
<h2 id="conclusion">
Conclusion
<a href="#conclusion" class="heading-anchor">#</a>
</h2>
<p>The field of protein design has a long history that dates back to the 1900s, when the first proteins were artificially synthesized. Work on directed evolution in the 1990s laid the foundation for the field, and the advent of high-throughput sequencing has made it possible to design proteins on a much larger scale. Today, protein design is an active area of research with many applications in various fields, including drug discovery, renewable energy, and synthetic biology. In part 2 of this series, we&rsquo;ll look at how machine learning is</p>
<h2 id="references">
References
<a href="#references" class="heading-anchor">#</a>
</h2>
<p>[1]:<br>
[2]: [Design by Directed Evolution]</p>
</section>
  <section><footer class="page-footer">
<hr />

<div class="previous-post" style="display:inline-block;">
  
  <a class="link-reverse" href="http://localhost:57852/post/2022-11-21-decentralization-of-atoms-is-underrated/?ref=footer">« Decentralization of Atoms is Underrated</a>
  
</div>

<div class="next-post", style="display:inline-block;float:right;">
  
  <a class="link-reverse" href="http://localhost:57852/post/2023-01-02-reflections-on-2022/?ref=footer">Reflections on 2022 »</a>
  
</div>

<ul class="page-footer-menu">
  
  
  <li><a href="https://twitter.com/an1lam">Twitter</a></li>
  
  
  

  
  <li><a href="https://github.com/an1lam">GitHub</a></li>
  

  

  

  

  

  

  

  

  

  

  
  <li><a href="https://scholar.google.com/citations?user=Q6_3PJEAAAAJ">Google Scholar</a></li>
  
  
  
    <li><a href="https://github.com/an1lam"><i class='fa fa-github fa-2x'></i>  </a></li>
  
    <li><a href="https://twitter.com/an1lam"><i class='fa fa-twitter fa-2x'></i>  </a></li>
  
    <li><a href="mailto:stephenmalina@gmail.com"><i class='fa fa-envelope fa-2x'></i>  </a></li>
  
    <li><a href="https://stackoverflow.com/users/1631855/an1lam"><i class='fa fa-stack-overflow fa-2x'></i>  </a></li>
  
</ul>


<p>
  Powered by <a href="https://gohugo.io">Hugo</a> and the
  <a href="https://github.com/loikein/hugo-tufte">Tufte theme</a>.
</p>




</footer>
</section>
  <section><nav class="menu">
    <ul>
    
        <li><a href="http://localhost:57852/">Home</a></li>
    
        <li><a href="http://localhost:57852/about">About</a></li>
    
        <li><a href="http://localhost:57852/post">Post</a></li>
    
        <li><a href="http://localhost:57852/bets">Bets</a></li>
    
        <li><a href="http://localhost:57852/greatest-hits">Greatest Hits</a></li>
    
        <li><a href="http://localhost:57852/tags">Tags</a></li>
    
    </ul>
</nav>
</section>
</article>





</body>

</html>
