<!DOCTYPE html>
<html lang="en-us"><head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1314&amp;path=livereload" data-no-instant defer></script>
<title>Paper Review - DeepSEA - Stephen Malina</title>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, viewport-fit=cover">
<meta name="description"
    content="A review of the DeepSEA paper, which uses convolutional neural networks to predict chromatin features from DNA sequences and evaluate the functional significance of non-coding variants. ">
<link rel="canonical" href="http://localhost:1314/post/2019-08-08-deepsea/" />



<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/modern-normalize/1.1.0/modern-normalize.min.css" crossorigin="anonymous" referrerpolicy="no-referrer" />



<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
<link rel="preload" as="style"
      href="https://fonts.googleapis.com/css?family=Noto+Serif+SC|Noto+Emoji&display=swap" />
<link rel="stylesheet"
      href="https://fonts.googleapis.com/css?family=Noto+Serif+SC|Noto+Emoji&display=swap"
      media="print" onload="this.media='all'" />
<noscript>
<link rel="stylesheet"
      href="https://fonts.googleapis.com/css?family=Noto+Serif+SC&display=swap" />
</noscript>


  
  
    
      <link rel="stylesheet" href="http://localhost:1314/css/hugo-tufte.min.css">
    
  



  
  
    
      <link rel="stylesheet" href="http://localhost:1314/css/hugo-tufte-options.min.css">
    
  


<link rel="stylesheet" href="http://localhost:1314/css/hugo-tufte-override.css">

</head>
<body>


<article id="main">
  <section>
<h1 class="content-title">Paper Review - DeepSEA</h1></section>

  

  <section><p>In which I record my thoughts on <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4768299/pdf/nihms757739.pdf">DeepSEA</a>.</p>
<h2 id="terminology">
Terminology
<a href="#terminology" class="heading-anchor">#</a>
</h2>
<ul>
<li>ChIP-seq (Chromatin Immunoprecipitation Sequencing):</li>
<li>Expression Quantitative Trait Locis (eQTLs)</li>
<li>Cofactor Binding Sequences</li>
<li>Histone Marks: modifications to histone proteins in the nucleosome that impact the shape of chromatin.</li>
<li>gkm-SVMs: The gkm-SVM is the previous SOTA model for predicting transcription factor binding based on ChIP-seq data.</li>
<li>Allele: variants of a gene at the same position on a chromosome.</li>
</ul>
<h2 id="what-is-this-paper-about">
What is this paper about?
<a href="#what-is-this-paper-about" class="heading-anchor">#</a>
</h2>
<p>This paper trains a CNN to predict the presence of 919 &ldquo;chromatin features&rdquo;&ndash;different TF binding sites, DHSs, and histone marks&ndash;from 1000bp DNA sequences. It then tests this model by using a functional significance score based on its output to train a logistic regression classifier to predict whether SNPs will be present in a few different catalog of SNPs known to impact different biological functions, e.g. a GWAS catalog of disease-related SNPs.</p>
<h2 id="technical-methods">
Technical Methods
<a href="#technical-methods" class="heading-anchor">#</a>
</h2>
<h2 id="why-is-this-important">
Why is this important?
<a href="#why-is-this-important" class="heading-anchor">#</a>
</h2>
<h2 id="meta">
Meta
<a href="#meta" class="heading-anchor">#</a>
</h2>
<p>This is the clearest of the three papers I&rsquo;ve read so far, but that may be because I&rsquo;m slowly internalizing the language of the domain.</p>
<h2 id="questions">
Questions
<a href="#questions" class="heading-anchor">#</a>
</h2>
<ul>
<li>
<p>What does this mean? I guess this paper precedes DeepBind so maybe this was just the first of its kind&hellip;</p>
<blockquote>
<p>Such approaches are valuable for prioritizing sequence variants;
however, current methods except for a parallel work have not been able to extract and
utilize regulatory sequence information de novo for noncoding-variant function prediction,
which requires precise allele-specific prediction with single-nucleotide sensitivity. In fact,
no previous approach predicts functional effects of noncoding variants from only genomic
sequence, and no method has been demonstrated to predict with single-nucleotide sensitivity
the effects of noncoding variants on transcription factor (TF) binding, DNA accessibility and
histone marks of sequences.</p></blockquote>
</li>
<li>
<p>What&rsquo;s the difference between &ldquo;motifs&rdquo; and &ldquo;evolutionary features and chromatin annotations&rdquo;?</p>
</li>
<li>
<p>When they trained the logistic regression classifiers, what aspects of the original model were they using? I think I understand, you can use the output of the NN model as the input to the classifier and train the classifier to predict based on it (predicted impact on chromatin shape).</p>
</li>
<li>
<p>What do the probability predictions in the output represent? Are they continuous or purely for classification? Is the impact on a chromatin feature binary? Yes, the outputs are probabilities but the actual thing we care about is a binary classification (either will the factor bind or will there be a histone mark).</p>
</li>
<li>
<p>Is the histone mark on the nucleosome being predicted by the DNA sequence or is the DNA sequence content causing the histone mark? It&rsquo;s believed to be the latter although there&rsquo;s apparently some work claiming that the former can also happen.</p>
</li>
</ul>
</section>
  <section><footer class="page-footer">
<hr />

<div class="previous-post" style="display:inline-block;">
  
  <a class="link-reverse" href="http://localhost:1314/post/2019-08-05-basset/?ref=footer">« Paper Review - Basset</a>
  
</div>

<div class="next-post", style="display:inline-block;float:right;">
  
  <a class="link-reverse" href="http://localhost:1314/post/2019-09-07-matrix-potpourri/?ref=footer">Matrix Potpourri »</a>
  
</div>

<ul class="page-footer-menu">
  
  
  <li><a href="https://twitter.com/an1lam">Twitter</a></li>
  
  
  

  
  <li><a href="https://github.com/an1lam">GitHub</a></li>
  

  

  

  

  

  

  

  

  

  

  
  <li><a href="https://scholar.google.com/citations?user=Q6_3PJEAAAAJ">Google Scholar</a></li>
  
  
  
    <li><a href="https://github.com/an1lam"><i class='fa fa-github fa-2x'></i>  </a></li>
  
    <li><a href="https://twitter.com/an1lam"><i class='fa fa-twitter fa-2x'></i>  </a></li>
  
    <li><a href="mailto:stephenmalina@gmail.com"><i class='fa fa-envelope fa-2x'></i>  </a></li>
  
    <li><a href="https://stackoverflow.com/users/1631855/an1lam"><i class='fa fa-stack-overflow fa-2x'></i>  </a></li>
  
</ul>


<p>
  Powered by <a href="https://gohugo.io">Hugo</a> and the
  <a href="https://github.com/loikein/hugo-tufte">Tufte theme</a>.
</p>




</footer>
</section>
  <section><nav class="menu">
    <ul>
    
        <li><a href="http://localhost:1314/">Home</a></li>
    
        <li><a href="http://localhost:1314/about">About</a></li>
    
        <li><a href="http://localhost:1314/post">Post</a></li>
    
        <li><a href="http://localhost:1314/bets">Bets</a></li>
    
        <li><a href="http://localhost:1314/greatest-hits">Greatest Hits</a></li>
    
        <li><a href="http://localhost:1314/tags">Tags</a></li>
    
    </ul>
</nav>
</section>
</article>







  <script>(function(){var e,t,n,s=document.getElementsByTagName("code");for(n=0;n<s.length;){if(t=s[n],t.parentNode.tagName!=="PRE"&&t.childElementCount===0&&!t.classList.contains("nolatex")&&(e=t.textContent,/^\$[^$]/.test(e)&&/[^$]\$$/.test(e)&&(e=e.replace(/^\$/,"\\(").replace(/\$$/,"\\)"),t.textContent=e),/^\\\((.|\s)+\\\)$/.test(e)||/^\\\[(.|\s)+\\\]$/.test(e)||/^\$(.|\s)+\$$/.test(e)||/^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(e))){t.outerHTML=t.innerHTML;continue}n++}})()</script>


<link rel="stylesheet" href="//cdn.jsdelivr.net/npm/katex@0.16.4/dist/katex.min.css" crossorigin="anonymous">
<script defer src="//cdn.jsdelivr.net/npm/katex@0.16.4/dist/katex.min.js" crossorigin="anonymous"></script>
<script defer src="//cdn.jsdelivr.net/npm/katex@0.16.4/dist/contrib/auto-render.min.js" crossorigin="anonymous"></script>

<script>
    document.addEventListener("DOMContentLoaded", function() {
        renderMathInElement(document.body, {
            
            
            
            
            
            
            trust: (context) => ['\\htmlId', '\\href'].includes(context.command),
            macros: {
              "\\eqref": "\\href{###1}{(\\text{#1})}",
              "\\ref": "\\href{###1}{\\text{#1}}",
              "\\label": "\\htmlId{#1}{}"
            }
        });
    });
</script>



</body>

</html>
