<!doctype html>
<html lang="en">
<head><meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta content='text/html; charset=utf-8' http-equiv='content-type' />

  <title>Paper Review - DeepSEA - Stephen Malina</title>
  <meta content='Paper Review - DeepSEA - Stephen Malina' property='title' />
  <meta content='Paper Review - DeepSEA - Stephen Malina' property='og:title' />


<meta property="og:description" content="In which I record my thoughts on DeepSEA.
Terminology  ChIP-seq (Chromatin Immunoprecipitation Sequencing): Expression Quantitative Trait Locis (eQTLs) Cofactor Binding Sequences Histone Marks: modifications to histone proteins in the nucleosome that impact the shape of chromatin. gkm-SVMs: The gkm-SVM is the previous SOTA model for predicting transcription factor binding based on ChIP-seq data. Allele: variants of a gene at the same position on a chromosome.  What is this paper about?" />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://an1lam.github.io/post/2019-08-08-deepsea/" />


<meta property="article:published_time" content="2019-08-08T00:00:00&#43;00:00"/>

<meta property="article:modified_time" content="2019-08-08T00:00:00&#43;00:00"/>








<meta name="generator" content="Hugo 0.57.2" />

<link href="https://fonts.googleapis.com/css?family=Source+Sans+Pro:400,600" rel="stylesheet">
<style type="text/css">/*https://coolors.co/afd5aa-f0f2ef-a69f98-3d3d3d-8c6057*/
:root {
  --main-color: #8C6056; 
  --secondary-color: #AFD5AA;
  --logo-text-color: #fff;
  --body-text-color: #3d3d3d;
  --heading-text-color: #383838;
  --background-color: #fff;
}</style>
<link href='/css/tachyons.min.css' rel="stylesheet">
<link href='/css/styles.css' rel="stylesheet">


<link rel="icon" 
 
  href='/favicon.ico'

type="image/x-icon"/>

<link href='/feed.xml' rel="alternate" type="application/atom+xml" title="Stephen Malina" />
</head>
<body class="global-font">
  <nav class=" flex-ns justify-between border-box pa3 pl3-l pr2-l mt1 mt0-ns" id="navbar">
  <div class="flex">
    <a class="f4 fw6 ttu no-underline dim bg-main-color pv1 ph2 br2" id="site-title" href='/' title="Home">Stephen Malina</a>
  </div>
  
  <div class=" flex-ns mt2 mt0-ns pv1">
    
      <a class="link dim dark-gray f6 dib mr2 mr3-l ttu tracked" href='/about' title="About">About</a>
    
      <a class="link dim dark-gray f6 dib mr2 mr3-l ttu tracked" href='/' title="Blog">Blog</a>
    
      <a class="link dim dark-gray f6 dib mr2 mr3-l ttu tracked" href='/progress' title="The State of My Mind">The State of My Mind</a>
    
  </div>
  
</nav>
  
<main class="center mv4 content-width ph3">
  <div class="f3 fw6 heading-color heading-font post-title">Paper Review - DeepSEA</div>
  <p class="silver f6 mt1 mb4 post-meta">
    <time>08 Aug 2019</time> 
     | 
    
    
    tags: [ <a href='/tags/review' class="link silver">review</a> <a href='/tags/machine-learning' class="link silver">machine-learning</a> <a href='/tags/genomics' class="link silver">genomics</a> <a href='/tags/paper' class="link silver">paper</a> <a href='/tags/notes' class="link silver">notes</a>  ]
    
  </p>
  <div class="lh-copy post-content">

<p>In which I record my thoughts on <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4768299/pdf/nihms757739.pdf">DeepSEA</a>.</p>

<h2 id="terminology">Terminology</h2>

<ul>
<li>ChIP-seq (Chromatin Immunoprecipitation Sequencing):</li>
<li>Expression Quantitative Trait Locis (eQTLs)</li>
<li>Cofactor Binding Sequences</li>
<li>Histone Marks: modifications to histone proteins in the nucleosome that impact the shape of chromatin.</li>
<li>gkm-SVMs: The gkm-SVM is the previous SOTA model for predicting transcription factor binding based on ChIP-seq data.</li>
<li>Allele: variants of a gene at the same position on a chromosome.</li>
</ul>

<h2 id="what-is-this-paper-about">What is this paper about?</h2>

<p>This paper trains a CNN to predict the presence of 919 &ldquo;chromatin features&rdquo;&ndash;different TF binding sites, DHSs, and histone marks&ndash;from 1000bp DNA sequences. It then tests this model by using a functional significance score based on its output to train a logistic regression classifier to predict whether SNPs will be present in a few different catalog of SNPs known to impact different biological functions, e.g. a GWAS catalog of disease-related SNPs.</p>

<h2 id="technical-methods">Technical Methods</h2>

<h2 id="why-is-this-important">Why is this important?</h2>

<h2 id="meta">Meta</h2>

<p>This is the clearest of the three papers I&rsquo;ve read so far, but that may be because I&rsquo;m slowly internalizing the language of the domain.</p>

<h2 id="questions">Questions</h2>

<ul>
<li><p>What does this mean? I guess this paper precedes DeepBind so maybe this was just the first of its kind&hellip;</p>

<blockquote>
<p>Such approaches are valuable for prioritizing sequence variants;
however, current methods except for a parallel work have not been able to extract and
utilize regulatory sequence information de novo for noncoding-variant function prediction,
which requires precise allele-specific prediction with single-nucleotide sensitivity. In fact,
no previous approach predicts functional effects of noncoding variants from only genomic
sequence, and no method has been demonstrated to predict with single-nucleotide sensitivity
the effects of noncoding variants on transcription factor (TF) binding, DNA accessibility and
histone marks of sequences.</p>
</blockquote></li>

<li><p>What&rsquo;s the difference between &ldquo;motifs&rdquo; and &ldquo;evolutionary features and chromatin annotations&rdquo;?</p></li>

<li><p>When they trained the logistic regression classifiers, what aspects of the original model were they using? I think I understand, you can use the output of the NN model as the input to the classifier and train the classifier to predict based on it (predicted impact on chromatin shape).</p></li>

<li><p>What do the probability predictions in the output represent? Are they continuous or purely for classification? Is the impact on a chromatin feature binary? Yes, the outputs are probabilities but the actual thing we care about is a binary classification (either will the factor bind or will there be a histone mark).</p></li>

<li><p>Is the histone mark on the nucleosome being predicted by the DNA sequence or is the DNA sequence content causing the histone mark? It&rsquo;s believed to be the latter although there&rsquo;s apparently some work claiming that the former can also happen.</p></li>
</ul>
</div>
</main>
 






<div class="tl fixed list-pages lh-copy" id="contents-list"></div>



<div class="pagination tc tr-l db fixed-l bottom-2-l right-2-l mb3 mb0-l">
  
<a id="scroll-to-top" class="f6 o-0 link br2 ph2 pv1 mb1 bg-main-color pointer" onclick="topFunction()" style="color: #fff; visibility: hidden; display: none; transition: opacity .5s, visibility .5s;" title="back to top">back to top</a>
<br>
  <p class="mb0 mt2">
  <a href="https://an1lam.github.io/post/2019-08-05-basset/">prev post</a>
  <a href="https://an1lam.github.io/post/2019-09-07-matrix-potpourri/">next post</a>
  </p>
</div>

  <footer class="content-width mt0 mt5-l mb4 f6 center ph3 gray tc tl-l">
  <hr class="dn db-l ml0-l gray w3"><br>
  Powered by <a href="https://gohugo.io/" target="_blank" class="link gray dim">Hugo</a>, based on the <a href="https://github.com/lingxz/er" target="_blank" class="link gray dim">Er</a> theme. <br>
  
</footer>
  



<script src="https://cdnjs.cloudflare.com/ajax/libs/tocbot/4.4.2/tocbot.min.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/tocbot/4.4.2/tocbot.css">
<style>.is-active-link::before { background-color: var(--secondary-color); }</style>




<script type="text/javascript">
var prevScrollpos = window.pageYOffset;
window.onscroll = function() {
  var currentScrollPos = window.pageYOffset;

  
  if (document.getElementById("tag-cloud") !== null) { 
    if (prevScrollpos > currentScrollPos) { 
      document.getElementById("tag-cloud").style.visibility = "visible";
      document.getElementById("tag-cloud").style.opacity = "1";
    } else {
      document.getElementById("tag-cloud").style.visibility = "hidden";
      document.getElementById("tag-cloud").style.opacity = "0";
    }
  }
  

  
  if (document.body.scrollTop > 1000 || document.documentElement.scrollTop > 1000) {
      document.getElementById("scroll-to-top").style.display = "inline";
      document.getElementById("scroll-to-top").style.visibility = "visible";
      document.getElementById("scroll-to-top").style.opacity = "1";
  } else {
      document.getElementById("scroll-to-top").style.visibility = "hidden";
      document.getElementById("scroll-to-top").style.opacity = "0";
  }
  
  prevScrollpos = currentScrollPos;
}


function topFunction() {
  document.body.scrollTop = 0; 
  document.documentElement.scrollTop = 0; 
}






if (document.getElementById("contents-list") !== null && document.getElementsByClassName("post-content").length !== 0) { 
  tocbot.init({
    
    tocSelector: '#contents-list',
    
    contentSelector: '.post-content',
    
    headingSelector: 'h1, h2, h3',
  });
}


</script>






<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [ ['$','$'], ["\\(","\\)"] ],
      processEscapes: true
    }
  });
</script>
<script type="text/javascript" async
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>


</body>
</html>