<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">

<head>
<title>Stephen Malina - Paper Review - DeepSEA</title>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="description" content="">
<meta name="keywords" content="">
<meta name="author" content="Stephen Malina">
<meta name="generator" content="Hugo 0.68.3" />

  
  

  <script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\\[','\\]']],
    processEscapes: true,
    processEnvironments: true,
    menuSettings: { zoom: "Double-Click" },
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript"
  src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>







<link rel="stylesheet" href="https://unpkg.com/purecss@2.0.3/build/pure-min.css">


    <link rel="stylesheet" href="https://unpkg.com/purecss@2.0.3/build/grids-responsive-min.css">








<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css">


<link rel="stylesheet" href="https://stephenmalina.com/css/tufte.css">
<link rel="stylesheet" href="https://stephenmalina.com/css/hugo-tufte.css">
<link rel="stylesheet" href="https://stephenmalina.com/css/hugo-tufte-override.css">

  
<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
	ga('create', 'UA-187780632-1', 'auto');
	
	ga('send', 'pageview');
}
</script>

  
<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
	ga('create', 'UA-187780632-1', 'auto');
	
	ga('send', 'pageview');
}
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>

</head>


<body>
<div id="layout" class="pure-g">
<article class="pure-u-1">
<header class="brand">
  <h1>Stephen Malina</h1>
  <h2>This is my blog. There are many others like it but this one is mine.</h2>
  <nav class="menu">
    <ul>
    
        <li><a href="https://stephenmalina.com/">Home</a></li>
    
        <li><a href="https://stephenmalina.com/about">About</a></li>
    
        <li><a href="https://stephenmalina.com/post">Post</a></li>
    
        <li><a href="https://stephenmalina.com/subscribe">Subscribe</a></li>
    
        <li><a href="https://stephenmalina.com/tags">Tags</a></li>
    
    </ul>
</nav>

</header>

<section>
<h1 class="content-title">
  
  <a href="https://stephenmalina.com/post/2019-08-08-deepsea/">Paper Review - DeepSEA</a>
  
</h1>



  <span class="content-meta">
    


    
      <i class="fa fa-calendar"></i>
      &nbsp;Aug 8, 2019
    

    
      &nbsp;<i class="fa fa-clock-o"></i>
      &nbsp;3 min read
    

    
  </span>


</section>



<section><p>In which I record my thoughts on <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4768299/pdf/nihms757739.pdf">DeepSEA</a>.</p>
<h2 id="terminology">Terminology</h2>
<ul>
<li>ChIP-seq (Chromatin Immunoprecipitation Sequencing):</li>
<li>Expression Quantitative Trait Locis (eQTLs)</li>
<li>Cofactor Binding Sequences</li>
<li>Histone Marks: modifications to histone proteins in the nucleosome that impact the shape of chromatin.</li>
<li>gkm-SVMs: The gkm-SVM is the previous SOTA model for predicting transcription factor binding based on ChIP-seq data.</li>
<li>Allele: variants of a gene at the same position on a chromosome.</li>
</ul>
<h2 id="what-is-this-paper-about">What is this paper about?</h2>
<p>This paper trains a CNN to predict the presence of 919 &ldquo;chromatin features&rdquo;&ndash;different TF binding sites, DHSs, and histone marks&ndash;from 1000bp DNA sequences. It then tests this model by using a functional significance score based on its output to train a logistic regression classifier to predict whether SNPs will be present in a few different catalog of SNPs known to impact different biological functions, e.g. a GWAS catalog of disease-related SNPs.</p>
<h2 id="technical-methods">Technical Methods</h2>
<h2 id="why-is-this-important">Why is this important?</h2>
<h2 id="meta">Meta</h2>
<p>This is the clearest of the three papers I&rsquo;ve read so far, but that may be because I&rsquo;m slowly internalizing the language of the domain.</p>
<h2 id="questions">Questions</h2>
<ul>
<li>
<p>What does this mean? I guess this paper precedes DeepBind so maybe this was just the first of its kind&hellip;</p>
<blockquote>
<p>Such approaches are valuable for prioritizing sequence variants;
however, current methods except for a parallel work have not been able to extract and
utilize regulatory sequence information de novo for noncoding-variant function prediction,
which requires precise allele-specific prediction with single-nucleotide sensitivity. In fact,
no previous approach predicts functional effects of noncoding variants from only genomic
sequence, and no method has been demonstrated to predict with single-nucleotide sensitivity
the effects of noncoding variants on transcription factor (TF) binding, DNA accessibility and
histone marks of sequences.</p>
</blockquote>
</li>
<li>
<p>What&rsquo;s the difference between &ldquo;motifs&rdquo; and &ldquo;evolutionary features and chromatin annotations&rdquo;?</p>
</li>
<li>
<p>When they trained the logistic regression classifiers, what aspects of the original model were they using? I think I understand, you can use the output of the NN model as the input to the classifier and train the classifier to predict based on it (predicted impact on chromatin shape).</p>
</li>
<li>
<p>What do the probability predictions in the output represent? Are they continuous or purely for classification? Is the impact on a chromatin feature binary? Yes, the outputs are probabilities but the actual thing we care about is a binary classification (either will the factor bind or will there be a histone mark).</p>
</li>
<li>
<p>Is the histone mark on the nucleosome being predicted by the DNA sequence or is the DNA sequence content causing the histone mark? It&rsquo;s believed to be the latter although there&rsquo;s apparently some work claiming that the former can also happen.</p>
</li>
</ul>
</section>
<section>
  

  <footer class="page-footer">
		<hr>
		<ul class="page-footer-menu">
		
      <li><a href="https://github.com/an1lam"><i class='fa fa-github fa-2x'></i> </a></li>
		
      <li><a href="https://twitter.com/an1lam"><i class='fa fa-twitter fa-2x'></i> </a></li>
		
      <li><a href="mailto:stephenmalina@gmail.com"><i class='fa fa-envelope fa-2x'></i> </a></li>
		
      <li><a href="https://stackoverflow.com/users/1631855/an1lam"><i class='fa fa-stack-overflow fa-2x'></i> </a></li>
		
		</ul>

  
    <p>
      Powered by <a href="https://gohugo.io">Hugo</a> and the
      <a href="https://github.com/shawnohare/hugo-tufte">Tufte theme</a>.
    </p>
  

	<div class="copyright">
	<p>
    
      &copy; 2021
    .
    All rights reserved.
    
  </p>
</div>
</footer>



</section>
</article>
</div>
</body>
</html>
