<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">

<head>
<title>Stephen Malina - Paper Review - Basset</title>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="description" content="">
<meta name="keywords" content="">
<meta name="author" content="Stephen Malina">
<meta name="generator" content="Hugo 0.40.1" />

  
  

  <script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\\[','\\]']],
    processEscapes: true,
    processEnvironments: true,
    menuSettings: { zoom: "Double-Click" },
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript"
  src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>







<link rel="stylesheet" href="https://unpkg.com/purecss@2.0.3/build/pure-min.css">


    <link rel="stylesheet" href="https://unpkg.com/purecss@2.0.3/build/grids-responsive-min.css">








<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css">


<link rel="stylesheet" href="https://stephenmalina.com/css/tufte.css">
<link rel="stylesheet" href="https://stephenmalina.com/css/hugo-tufte.css">
<link rel="stylesheet" href="https://stephenmalina.com/css/hugo-tufte-override.css">

</head>


<body>
<div id="layout" class="pure-g">
<article class="pure-u-1">
<header class="brand">
  <h1>Stephen Malina</h1>
  <h2>This is my blog. There are many others like it but this one is mine.</h2>
  <nav class="menu">
    <ul>
    
        <li><a href="https://stephenmalina.com/">Home</a></li>
    
        <li><a href="https://stephenmalina.com/about">About</a></li>
    
        <li><a href="https://stephenmalina.com/post">Post</a></li>
    
        <li><a href="https://stephenmalina.com/subscribe">Subscribe</a></li>
    
        <li><a href="https://stephenmalina.com/tags">Tags</a></li>
    
    </ul>
</nav>

</header>

<section>
<h1 class="content-title">
  
  <a href="https://stephenmalina.com/post/2019-08-05-basset/">Paper Review - Basset</a>
  
</h1>



  <span class="content-meta">
    


    
      <i class="fa fa-calendar"></i>
      &nbsp;Aug 5, 2019
    

    
      &nbsp;<i class="fa fa-clock-o"></i>
      &nbsp;5 min read
    

    
  </span>


</section>



<section>

<p>In which I record my thoughts on <a href="https://genome.cshlp.org/content/26/7/990.full">Basset</a>.</p>

<h2 id="bio-background">Bio Background</h2>

<p>The genome consists of (broadly) two types of genes, coding genes and noncoding genes. Coding genes get translated into proteins (as laid down by the Central Dogma) and are what most of us learned about in bio class. Non-coding genes&hellip; don&rsquo;t. As I understand it, noncoding genes can do a bunch of different things, but part of their function is to regulate coding gene activity through a number of mechanisms.</p>

<p>Related to this, in the genomics world, there&rsquo;s a concept of &ldquo;DNA accessibility&rdquo;. In eukaryotic cells in particular, DNA is intertwined into a material (or structure, not sure what the right term is here) called chromatin. This chromatin can take on different shapes, which impact which genes are accessed by transcription (become proteins). If I&rsquo;m understanding correctly, certain noncoding genes can impact chromatin shape and thereby transcription.</p>

<h2 id="what-is-this-paper-about">What is this paper about?</h2>

<p>This paper focuses on predicting DNA sequence accessibility in different cell types with sequences as input and the output from DNase-seq as training data. They use the output of their trained model to re-discover sequences that are already known to correlate with cell-specific accessibility and use it to do what they call &ldquo;in-silico saturation mutagenesis&rdquo; - briefly, testing the impact of single nucleotide mutations on predicted accessibility.</p>

<h2 id="technical-methods">Technical Methods</h2>

<p>The Basset model is a classifier that takes a one-hot encoded DNA sequence as input at test time and outputs accessibility classifications (accessible or not) for 164 different cell types. At training time, the model takes the sequence plus a binary vector of accessibility for the 164 different cell types.</p>

<p>Basset is a 6-layer convolutional neural network. 3 of the layers are convolutional layers and use max pooling, batch norm, and rectification. 2 subsequent layers use ReLu neurons with dropout. And a final sigmoid neuron layer has 164 neurons representing accessibility for the 164 different cell types.</p>

<p>The authors used Bayesian optimization for hyperparameter tuning during training.</p>

<p>They did something fairly interesting on the interpretability front, which I&rsquo;m still trying to wrap my head around. (I&rsquo;m not sure if this is common.) Based on the assumption that convolutional filters each would capture knowledge about accessibility-related sequence motifs, they nullified filters one by one and observed the change in outputs (by comparing the sum of squares of the net&rsquo;s output before and after). They then clustered filters by their impact on accessibility predictions revealed a set of filters that matched known factors involved in epithethial development.</p>

<p>They also did something else cool where they converted filters to PWMs and then compared them to known transcription factor motifs using a tool called TomTom. This allowed them to see whether filters actually captured &ldquo;real features&rdquo; of the data. Combined with the previous method, this also enables them and others to look for potentially rare but important sequences that impact accessibility.</p>

<h2 id="why-is-this-important">Why is this important?</h2>

<p>Through reading this paper, I think I got a better sense of what aspects of these models might actually be useful than I had <a href="https://stephenmalina.com/content/post/2019-07-26-deepbind">when I read about DeepBind</a>.</p>

<p>I was told this by my professor, but didn&rsquo;t fully grok it until now - biologists care much less about prediction than ML people do and much more about interpretability. We see that in this paper - most of the predictions the model makes have already been validated in experimental studies. The authors discuss a few leads to follow, but I suspect it&rsquo;s hard to convince other biologists to do the work to follow them.</p>

<p>On the other hand, the way the authors interrogated the model and found meaningful information inside of its filters is pretty exciting to me. It seems like comp. bio / ML people have an opportunity to explore interpreting neural nets in ways that make less sense in the world of image processing and NLP.</p>

<p>Related to this, I&rsquo;m excited by and think this line of research is valuable for two primary reasons. First, to the extent it moves us towards a world where ML models can actually be used to guide/substitute for expensive experimental work to discover noncoding sequence/accessibility relationships, it promises to make genomics research more efficient. Second, I have this vision of ML being used to build better knowledge maps of biology that enable biologists to answer questions by actually probing models and using them almost like expert systems. The interpretability stuff in this paper seems like a step in that direction.</p>

<h2 id="meta">Meta</h2>

<p>So far, I&rsquo;m frustrated by how bad the definitions and descriptions of biological methods / processes are in these papers. Maybe I don&rsquo;t realize the same thing is true in CS, but I repeatedly find myself wanting to answer a simple question like &ldquo;what is DNase-seq?&rdquo; and, instead of getting a simple description, finding detailed discussions of the processes. As a Computer Scientist, I wish biologists described things in terms of abstractions more and mechanisms less. For example, why can&rsquo;t any of these sources just tell me what the output number from DNase-seq (or any other process) means? I lost hours just trying to figure out what the output values of Basset&rsquo;s NN represented and trying to understand what geneticists mean by &ldquo;enrichment&rdquo;.</p>

<p>In case anyone else has this problem, a tip is that I&rsquo;ve ironically found that file formats and model inputs are easier to decipher than the prose descriptions in some cases.</p>

<h2 id="questions">Questions</h2>

<ul>
<li>Are DHSs DNA sequences or something else? (Tenative answer) Yes, I think at least in the context of this paper, we care about sequence specific DNase enzymes.</li>
<li>Why is accessibility binary? Is it? No, it&rsquo;s not. The NN seems to be outputting a continuous measure of accessibility. That said, I don&rsquo;t know what the unit of chromatin accessibility actually is&hellip;</li>
</ul>
</section>
<section>
  

  <footer class="page-footer">
		<hr>
		<ul class="page-footer-menu">
		
      <li><a href="https://github.com/an1lam"><i class='fa fa-github fa-2x'></i> </a></li>
		
      <li><a href="https://twitter.com/an1lam"><i class='fa fa-twitter fa-2x'></i> </a></li>
		
      <li><a href="mailto:stephenmalina@gmail.com"><i class='fa fa-envelope fa-2x'></i> </a></li>
		
      <li><a href="https://stackoverflow.com/users/1631855/an1lam"><i class='fa fa-stack-overflow fa-2x'></i> </a></li>
		
		</ul>

  
    <p>
      Powered by <a href="https://gohugo.io">Hugo</a> and the
      <a href="https://github.com/shawnohare/hugo-tufte">Tufte theme</a>.
    </p>
  

	<div class="copyright">
	<p>
    
      &copy; 2020
    .
    All rights reserved.
    
  </p>
</div>
</footer>



</section>
</article>
</div>
</body>
</html>
